frontend/src/pages/chat/Chat.tsx

```diff
@@ -32,26 +34,64 @@ const Chat = () => {
         lastQuestionRef.current = question;
 
         setIsLoading(true);
+        setShowLoadingMessage(true);
         const abortController = new AbortController();
         abortFuncs.current.unshift(abortController);
 
         const userMessage: ChatMessage = {
             role: "user",
-            content: question,
-            end_turn: null
+            content: question
         };
 
         const request: ConversationRequest = {
             messages: [...answers, userMessage]
         };
 
+        let result = {} as ChatResponse;
         try {
-            const result = await conversationApi(request, abortController.signal);
-            setAnswers([...answers, userMessage, ...result.choices[0].messages]);
-        } catch {
+            const response = await conversationApi(request, abortController.signal);
+            if (response?.body) {
+                
+                const reader = response.body.getReader();
+                let runningText = "";
+                while (true) {
+                    const {done, value} = await reader.read();
+                    if (done) break;
+
+                    var text = new TextDecoder("utf-8").decode(value);
+                    console.log(text);
+                    const objects = text.split("}{");
+                    objects.forEach((obj, index) => {
+                        if (obj[0] !== "{" && index !== 0) {
+                            obj = "{" + obj;
+                        }
+                        if (obj[obj.length - 1] !== "}" && index !== objects.length - 1) {
+                            obj = obj + "}";
+                        }
+                        try {
+                            runningText += obj;
+                            result = JSON.parse(runningText);
+                            setShowLoadingMessage(false);
+                            setAnswers([...answers, userMessage, ...result.choices[0].messages]);
```

Comment:
If everyone only looks at choices[0], do we need it?

Reply:
I wanted to stick to the API response format in case we do get multiple choices eventually